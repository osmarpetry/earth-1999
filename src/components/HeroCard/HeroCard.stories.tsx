import React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';

import HeroCard, { HeroCardProps } from 'components/HeroCard';

export default {
  title: 'Components/HeroCard',
  component: HeroCard,
} as Meta;

const Template: Story<HeroCardProps> = (args) => <HeroCard {...args} />;

export const Primary = Template.bind({});

Primary.args = {
  name: 'Spider-Man Playstation',
  imageSrc:
    'https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcQxR0m6S2SOdVD-rDfs88qcZJ-Ti44ScGJgeg&usqp=CAU',
  width: '221px',
  height: '221px',
};
