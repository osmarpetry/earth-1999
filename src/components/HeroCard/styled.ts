import styled from 'styled-components'

import colors from 'core/assets/styles/colors'
import fonts from 'core/assets/styles/fonts'

export const HeroCardStyled = styled.section<{ width: string }>`
  padding: 0;
  width: ${({ width }) => width};
  height: fit-content;
`

export const Image = styled.img<{ width: string; height: string }>`
  border-bottom: 3px solid red;
  width: ${({ width }) => width};
  height: ${({ height }) => height};
`

export const CardFooter = styled.footer`
  display: flex;
  justify-content: space-between;
  margin: 10px 0;
`

export const HeroName = styled.p`
  margin: 0;
  color: ${colors.fontPrimary};
  font-weight: ${fonts.lightBold};
`
