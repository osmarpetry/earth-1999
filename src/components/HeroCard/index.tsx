import React from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
import ContentLoader from 'react-content-loader';

import ButtonHeart, { HeartIcon } from 'components/ButtonHeart';

import colors from 'core/assets/styles/colors';
import fonts from 'core/assets/styles/fonts';
import sizes from 'core/assets/styles/sizes';

const HeroCardStyled = styled.section<{ width: string }>`
  padding: 0;
  width: ${({ width }) => width};
  height: fit-content;
`;

const Image = styled.img<{ width: string; height: string }>`
  border-bottom: 3px solid red;
  width: ${({ width }) => width};
  height: ${({ height }) => height};
`;

const CardFooter = styled.footer`
  display: flex;
  justify-content: space-between;
  margin: 10px 0;
`;

const HeroName = styled.p`
  margin: 0;
  color: ${colors.fontPrimary};
  font-weight: ${fonts.lightBold};
`;

export interface HeroCardProps {
  alt?: string;
  disabled?: boolean;
  favorite?: boolean;
  height: string;
  imageSrc?: string;
  linkTo?: string;
  loading?: boolean;
  name?: string;
  width: string;
  onClick?: () => void;
}

export default function HeroCard({
  alt = '',
  disabled = false,
  favorite = false,
  height,
  imageSrc = '',
  linkTo,
  loading = false,
  name = '',
  width,
  onClick = () => {},
}: HeroCardProps) {
  if (loading) {
    return (
      <HeroCardStyled width={width}>
        <ContentLoader
          width={width}
          height={height}
          style={{ borderRadius: '3px' }}
        >
          <rect width="100%" height="100%" />
        </ContentLoader>
        <CardFooter>
          <ContentLoader
            height={sizes.lineHeightParagraph}
            style={{ borderRadius: '3px', paddingRight: '10px' }}
          >
            <rect width="100%" height="100%" />
          </ContentLoader>
          <span
            style={{
              display: 'flex',
              justifyContent: 'center',
              width: 'fit-content',
            }}
          >
            <HeartIcon fill="rgba(255, 0,0,0.6)" stroke={"rgba(255, 0,0,0.6)"} />
          </span>
        </CardFooter>
      </HeroCardStyled>
    );
  }

  return (
    <HeroCardStyled width={width}>
      {linkTo ? (
        <Link to={linkTo}>
          <Image alt={alt} src={imageSrc} height={height} width={width} />
        </Link>
      ) : (
        <Image alt={alt} src={imageSrc} height={height} width={width} />
      )}
      <CardFooter>
        <HeroName>{name}</HeroName>
        <ButtonHeart value={favorite} disabled={disabled} onClick={onClick} />
      </CardFooter>
    </HeroCardStyled>
  );
}
