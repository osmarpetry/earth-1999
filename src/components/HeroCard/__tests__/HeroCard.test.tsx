import React from 'react'
import '@testing-library/jest-dom'
import { cleanup, fireEvent, render } from '@testing-library/react'

import renderWithRouter from 'core/utils/tests/renderWithRouter'

import HeroCard from 'components/HeroCard'

afterEach(cleanup)

it('should match HeroCard snapshot', () => {
  expect(
    renderWithRouter(
      <HeroCard
        loading={false}
        onClick={jest.fn}
        alt='Image alt'
        disabled={false}
        favorite={true}
        height='200px'
        width='170px'
        imageSrc='image.png'
        linkTo='/myHero'
        name='Hero'
      />
    )
  ).toMatchSnapshot()
})

it('should trigger the onClick event when input is clicked', () => {
  const onClick = jest.fn()
  const { getByTestId } = render(<HeroCard onClick={onClick} />)

  expect(onClick).not.toBeCalled()
  fireEvent.click(getByTestId('button-heart'))
  expect(onClick).toBeCalledTimes(1)
})

it('should render the alt and the name', () => {
  const { getByAltText, getByText } = render(
    <HeroCard alt='My alt' name='Hero name' />
  )

  expect(getByAltText('My alt')).toBeTruthy()
  expect(getByText('Hero name')).toBeTruthy()
})
