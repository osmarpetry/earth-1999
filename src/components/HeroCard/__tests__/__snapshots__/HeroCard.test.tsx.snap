// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match HeroCard snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="sc-gsTCUz ijBvsX"
        width="170px"
      >
        <a
          href="/myHero"
        >
          <img
            alt="Image alt"
            class="sc-dlfnbm iQbnro"
            height="200px"
            src="image.png"
            width="170px"
          />
        </a>
        <footer
          class="sc-hKgILt gGOuXv"
        >
          <p
            class="sc-eCssSg fvVVEn"
          >
            Hero
          </p>
          <button
            class="sc-bdfBwQ jowgqz"
            color="#FF1510"
            data-testid="button-heart"
          >
            <svg
              height="20px"
              version="1.1"
              viewBox="0 0 20 20"
              width="20px"
            >
              <title>
                Path Copy 2@1,5x
              </title>
              <g
                fill="none"
                fill-rule="evenodd"
                id="Page-1"
                stroke="none"
                stroke-width="1"
              >
                <g
                  class="teste"
                  fill="#FF1510"
                  fill-rule="nonzero"
                  id="Guide"
                  stroke="#FF0000"
                  stroke-width="2"
                  transform="translate(-231.000000, -828.000000)"
                >
                  <path
                    d="M248.428712,839.053345 L241.268331,846.885348 C241.199896,846.960594 241.102641,847 241.001802,847 C240.900941,847 240.803707,846.960594 240.735272,846.885348 L233.574892,839.053345 C231.475036,836.760342 231.475036,833.030654 233.574892,830.737651 C234.59779,829.616232 235.962887,829 237.418006,829 C238.750677,829 240.011307,829.515925 241.001802,830.465354 C241.992297,829.515925 243.249322,829 244.581993,829 C246.037112,829 247.405792,829.616232 248.432318,830.737651 C250.524963,833.030654 250.521357,836.760342 248.428712,839.053345 Z"
                    id="Path-Copy-2"
                  />
                </g>
              </g>
            </svg>
          </button>
        </footer>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="sc-gsTCUz ijBvsX"
      width="170px"
    >
      <a
        href="/myHero"
      >
        <img
          alt="Image alt"
          class="sc-dlfnbm iQbnro"
          height="200px"
          src="image.png"
          width="170px"
        />
      </a>
      <footer
        class="sc-hKgILt gGOuXv"
      >
        <p
          class="sc-eCssSg fvVVEn"
        >
          Hero
        </p>
        <button
          class="sc-bdfBwQ jowgqz"
          color="#FF1510"
          data-testid="button-heart"
        >
          <svg
            height="20px"
            version="1.1"
            viewBox="0 0 20 20"
            width="20px"
          >
            <title>
              Path Copy 2@1,5x
            </title>
            <g
              fill="none"
              fill-rule="evenodd"
              id="Page-1"
              stroke="none"
              stroke-width="1"
            >
              <g
                class="teste"
                fill="#FF1510"
                fill-rule="nonzero"
                id="Guide"
                stroke="#FF0000"
                stroke-width="2"
                transform="translate(-231.000000, -828.000000)"
              >
                <path
                  d="M248.428712,839.053345 L241.268331,846.885348 C241.199896,846.960594 241.102641,847 241.001802,847 C240.900941,847 240.803707,846.960594 240.735272,846.885348 L233.574892,839.053345 C231.475036,836.760342 231.475036,833.030654 233.574892,830.737651 C234.59779,829.616232 235.962887,829 237.418006,829 C238.750677,829 240.011307,829.515925 241.001802,830.465354 C241.992297,829.515925 243.249322,829 244.581993,829 C246.037112,829 247.405792,829.616232 248.432318,830.737651 C250.524963,833.030654 250.521357,836.760342 248.428712,839.053345 Z"
                  id="Path-Copy-2"
                />
              </g>
            </g>
          </svg>
        </button>
      </footer>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
